modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/DocumentMetadataSetResourceTest.java

```diff
@@ -124,22 +125,11 @@ private DocumentMetadataSet _addDocumentMetadataSet(Group group)
 					() -> LocaleUtil.toW3cLanguageIds(
 						ddmStructure.getAvailableLanguageIds()));
 				setDataDefinitionFields(
-					() -> {
-						DataDefinitionResource.Builder
-							dataDefinitionResourceBuilder =
-								_dataDefinitionResourceFactory.create();
-
-						DataDefinitionResource dataDefinitionResource =
-							dataDefinitionResourceBuilder.user(
-								TestPropsValues.getUser()
-							).build();
-
-						DataDefinition dataDefinition =
-							dataDefinitionResource.getDataDefinition(
-								ddmStructure.getStructureId());
-
-						return dataDefinition.getDataDefinitionFields();
-					});
+					() -> DataDefinitionFieldSerDes.toDTOs(
+						StringBundler.concat(
+							StringPool.OPEN_BRACKET,
+							ddmStructure.getDefinition(),
+							StringPool.CLOSE_BRACKET)));
```

Comment:
Not sure about this. If you look at com.liferay.data.engine.rest.internal.resource.v2_0.DataDefinitionResourceImpl#_normalizeDataDefinitionFields, there's some additional processing they do to the ddm structure when converting to data engine fields. Can you double check it is correct?